File: WebViewJavascriptInterafce/src/main/java/tech/thdev/webviewjavascriptinterface/webkit/CustomWebView.java
package tech.thdev.webviewjavascriptinterface.webkit;

import android.annotation.TargetApi;
import android.content.Context;
import android.os.Build;
import android.util.AttributeSet;
import android.webkit.WebSettings;
import android.webkit.WebView;

/**
 * Created by Tae-hwan on 8/4/16.
 */

public class CustomWebView extends WebView {

    public CustomWebView(Context context) {
        super(context);
    }

    public CustomWebView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public CustomWebView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    public CustomWebView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    public void init() {
        WebSettings settings = getSettings();
        settings.setJavaScriptEnabled(true);
        settings.setBuiltInZoomControls(true);
        settings.setLoadWithOverviewMode(true);
        settings.setUseWideViewPort(true);
        settings.setDisplayZoomControls(false);
        settings.setSupportZoom(false);
        settings.setDefaultTextEncodingName("UTF-8");

        // Setting Local Storage
        settings.setDatabaseEnabled(true);
        settings.setDomStorageEnabled(true);

        // No Cache
        settings.setCacheMode(WebSettings.LOAD_NO_CACHE);
    }
}

File: app/src/main/java/tech/thdev/app/webkit/CustomWebView.kt
package tech.thdev.app.webkit

import android.content.Context
import android.util.AttributeSet
import android.webkit.WebSettings
import android.webkit.WebView

/**
 * Created by Tae-hwan on 8/4/16.
 */
class CustomWebView @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleArr: Int = 0
) : WebView(context, attrs, defStyleArr) {

    fun init() {
        settings.javaScriptEnabled = true
        settings.builtInZoomControls = true
        settings.loadWithOverviewMode = true
        settings.useWideViewPort = true
        settings.displayZoomControls = false
        settings.setSupportZoom(false)
        settings.defaultTextEncodingName = "UTF-8"

        // Setting Local Storage
        settings.databaseEnabled = true
        settings.domStorageEnabled = true

        // No Cache
        settings.cacheMode = WebSettings.LOAD_NO_CACHE
    }
}